packages/caver-wallet/src/keyring/keyring.js

```diff
@@ -0,0 +1,845 @@
+            const roledKey = this._key[i]
+            if (i > KEY_ROLE.ROLE_TRANSACTION_KEY && roledKey.length > 0) isRoleBased = true
+            keyring.push(encryptKey(roledKey, password, options))
+        }
+
+        if (!isRoleBased) keyring = keyring[0]
+
+        return formatEncrypted(4, this._address, keyring, options)
+    }
+
+    /**
+     * encrypts a keyring and returns a keystore v3 object.
+     *
+     * @param {string} password The password to be used for keyring encryption. The encrypted key store can be decrypted with this password.
+     * @param {object} options The options to use when encrypt a keyring. See `keyring.encrypt` for more detail about options.
+     * @return {object}
+     */
+    encryptV3(password, options) {
+        const notAvailableError = `This keyring cannot be encrypted keystore v3. use 'keyring.encrypt(password)'.`
+        if (this._key[KEY_ROLE.ROLE_TRANSACTION_KEY].length > 1) throw new Error(notAvailableError)
+
+        for (let i = KEY_ROLE.ROLE_ACCOUNT_UPDATE_KEY; i < KEY_ROLE.ROLE_LAST; i++) {
+            if (this._key[i].length > 0) {
+                throw new Error(notAvailableError)
+            }
+        }
+
+        options = options || {}
+
+        const crypto = encryptKey(this._key[0][0], password, options)
+
+        return formatEncrypted(3, this._address, crypto, options)
+    }
+
+    _validateOptionsForUpdate(options = []) {
+        // { threshold: 1, weight: [1, 1] } => [{ threshold: 1, weight: [1, 1] }]
+        if (!_.isArray(options)) options = [options]
+
+        for (let i = 0; i < this._key.length; i++) {
+            // Validation for options obejct will be operated in AccountKeyWeightedMultiSig class.
+            if (options[i] && Object.keys(options[i]).length > 0) continue
+
+            let optionToAdd
+            if (this._key[i].length > 1) {
+                // default option when option is not set
+                optionToAdd = { threshold: 1, weight: Array(this._key[i].length).fill(1) }
+            } else {
+                // AccountKeyPublic does not need option
+                optionToAdd = {}
+            }
+
+            if (options[i]) {
+                options[i] = optionToAdd
+            } else {
+                options.push(optionToAdd)
+            }
+        }
+        return options
+    }
+}
+
+Keyring.PrivateKey = PrivateKey
+Keyring.role = KEY_ROLE
+
+module.exports = Keyring
+
+// PrivateKey instance: PrivateKey{}
+// single private key string: `0x{private key}`
+// multiple private key strings: [`0x{private key}`, `0x{private key}`]
+// multiple PrivateKey instances: [PrivateKey{}, PrivateKey{}]
+// role-based private key strings: [[`0x{private key}`], [`0x{private key}`, `0x{private key}`], [`0x{private key}`]]
+// role-based PrivateKey instances: [[PrivateKey{}], [PrivateKey{}, PrivateKey{}], [PrivateKey{}]]
+function formattingForKeyInKeyring(keyInput) {
```

Comment:
I think we need a description of this function. What is the return value? I think the above message is the input of this function.

Reply:
I added description for this function.